src/View/Helper/EncoreHelper.php

```diff
@@ -102,13 +102,25 @@ function (string $path): string {
      * Get js assets.
      *
      * @param string $asset The assets name.
+     * @param array $options Array of options and HTML arguments.
      * @return string HTML to load JS resources.
      */
-    public function script(string $asset): string
+    public function script(string $asset, array $options = []): string
     {
+        [$plugin, $resource] = pluginSplit($asset);
+        $map = $this->loadEntrypoints($plugin);
+        if ($map === null) {
+            return '';
+        }
+
+        $format = Hash::get($map, sprintf('entrypoints.%s.format', $resource), 'umd');
```

Comment:
I don't really like that pretty much the same logic is repeated both here and in `getAssets()`.

I'd prefer to either:
1. change the return type of `getAssets()` and always return a tuple `[string, string[]]`, e.g. `['esm', ['foo.js', 'bar.js']]` — although, being a public method, changing its return type would theoretically introduce a breaking change — or
2. extract a common `protected function getResourceInfo(string $asset): array` method that returns `Hash::get($map, ['entrypoints', $resource], [])`, and both this method and `getAssets()` use its return value to retrieve the key they're interested in

Reply:
I chose the first one :)